arangod/RocksDBEngine/RocksDBMethods.cpp

```diff
@@ -0,0 +1,94 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2023 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     http://www.apache.org/licenses/LICENSE-2.0
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Simon Grätzer
+////////////////////////////////////////////////////////////////////////////////
+
+#include "RocksDBMethods.h"
+
+#include <algorithm>
+#include <string>
+
+namespace arangodb {
+namespace {
+// size of std::string's internal SSO buffer.. only strings
+// that exceed this buffer require a dynamic memory allocation.
+// note: the size of the SSO buffer is implementation-defined
+// and is not guaranteed to be 15. we checked the SSO buffer
+// size for all current relevant implementation though:
+// - g++-11: 15
+// - g++-12: 15
+// - MSVC: 15
+// - clang++-14: 15
+// TODO: replace this constant with the expression
+//   std::string{}.capacity();
+// once all compilers provide constexpr std::string::capacity
+// implementations (clang 14 does, gcc 12 does, but gcc 11 doesn't).
```

Comment:
// gnu libstdc++ -- 15
// msvc stl -- 15
// llvm libc++ -- 22

The interesting moment it's easy to detect what implementation are used by their sizeof
sizeof(std::string)
// gnu libstdc++ -- 32
// msvc stl -- 32
// llvm libc++ -- 24